core/types/receipt.go

```diff
@@ -372,6 +371,9 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			rs[i].Logs[j].Index = logIndex
 			logIndex++
 		}
+		// also derive the Bloom if not derived yet
+		rs[i].Bloom = CreateBloom(Receipts{(*Receipt)(rs[i])})
```

Comment:
nitpick, the `*Receipt` is redundant.

`rs[i].Bloom = CreateBloom(Receipts{rs[i]})` is enough